### ISSUE TITLE
[ BUG ] TypeError: react_native_css_interop_1.StyleSheet.getFlag is not a function (it is undefined), js engine: hermes #224

### ISSUE THREAD

**Issue by periakteon** [nativewind/nativewind#1006]
**Describe the bug**
When I want to change the theme using the `ThemeToggle` component, I get the following error:

`TypeError: react_native_css_interop_1.StyleSheet.getFlag is not a function (it is undefined), js engine: hermes`

**To Reproduce**
1. Install the Nativewind package,
2. Create a root layout component like the one below,
3. Create ThemeToggle component

**Expected behavior**
The theme should change smoothly.

**Screenshots**
![image](https://github.com/user-attachments/assets/fe01085f-98e8-4153-873e-bb60589c3111)

https://github.com/user-attachments/assets/97300852-4384-467c-87ef-6ec216d7f111

**Platform:**
 - Type: [Emulator]
 - OS: [e.g. Android]

**Additional context**

`package.json`:

```
{
  "name": "@periakteon/hesap-lutfen",
  "main": "index.js",
  "version": "1.0.0",
  "scripts": {
    "start": "expo start --clear",
    "devbuild-android": "expo run:android",
    "devbuild-ios": "expo run:ios",
    "prebuild:android": "expo prebuild --platform android --clean",
    "prebuild:ios": "expo prebuild --platform ios --clean",
    "clean": "rm -rf .expo android node_modules",
    "lint": "eslint .",
    "android": "expo run:android",
    "ios": "expo run:ios"
  },
  "dependencies": {
    "@gorhom/bottom-sheet": "4",
    "@hookform/resolvers": "^3.9.0",
    "@react-native-async-storage/async-storage": "^2.0.0",
    "@react-navigation/native": "^6.0.2",
    "@rn-primitives/avatar": "~1.0.3",
    "@rn-primitives/label": "^1.0.3",
    "@rn-primitives/popover": "^1.0.3",
    "@rn-primitives/portal": "~1.0.3",
    "@rn-primitives/progress": "~1.0.3",
    "@rn-primitives/select": "^1.0.4",
    "@rn-primitives/separator": "^1.0.3",
    "@rn-primitives/slot": "~1.0.3",
    "@rn-primitives/switch": "^1.0.3",
    "@rn-primitives/tooltip": "~1.0.3",
    "@rn-primitives/types": "~1.0.3",
    "@rnmapbox/maps": "^10.1.28",
    "@shopify/flash-list": "1.6.4",
    "@supabase/supabase-js": "^2.45.1",
    "@tanstack/react-query": "^5.52.0",
    "@turf/distance": "^7.1.0",
    "@turf/helpers": "^7.1.0",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.1.0",
    "expo": "^51.0.24",
    "expo-dev-client": "~4.0.21",
    "expo-font": "~12.0.9",
    "expo-haptics": "~13.0.1",
    "expo-linking": "~6.3.1",
    "expo-location": "~17.0.1",
    "expo-navigation-bar": "~3.0.7",
    "expo-router": "~3.5.20",
    "expo-splash-screen": "~0.27.5",
    "expo-status-bar": "~1.12.1",
    "expo-system-ui": "~3.0.7",
    "lodash": "^4.17.21",
    "lottie-react-native": "^6.7.2",
    "lucide-react-native": "^0.378.0",
    "nativewind": "^4.0.33",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.52.2",
    "react-native": "0.74.5",
    "react-native-awesome-slider": "^2.5.3",
    "react-native-gesture-handler": "~2.16.1",
    "react-native-mmkv": "^2.12.2",
    "react-native-reanimated": "~3.10.1",
    "react-native-safe-area-context": "4.10.5",
    "react-native-screens": "~3.31.1",
    "react-native-svg": "15.2.0",
    "react-native-url-polyfill": "^2.0.0",
    "react-native-uuid": "^2.0.2",
    "react-native-web": "~0.19.6",
    "tailwind-merge": "^2.2.1",
    "tailwindcss": "3.3.5",
    "tailwindcss-animate": "^1.0.7",
    "zod": "^3.23.8",
    "zustand": "^4.5.5"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@types/lodash": "^4.17.7",
    "@types/react": "~18.2.79",
    "eslint": "^8.57.0",
    "eslint-config-expo": "^7.1.2",
    "typescript": "^5.3.3"
  },
  "private": true
}
```

---

`babel.config.js`:

```js
module.exports = function (api) {
  api.cache(true);
  return {
    presets: [
      ["babel-preset-expo", { jsxImportSource: "nativewind" }],
      "nativewind/babel",
    ],
    plugins: ["react-native-reanimated/plugin"],
  };
};
```

---

`tailwind.config.js`:

```js
const { hairlineWidth } = require("nativewind/theme");

// Color palette
// https://colorhunt.co/palette/ff204ea0153e5d0e4100224d

/** @type {import('tailwindcss').Config} */
module.exports = {
  darkMode: "class",
  content: ["./app/**/*.{ts,tsx}", "./components/**/*.{ts,tsx}"],
  presets: [require("nativewind/preset")],
  theme: {
    extend: {
      fontFamily: {
        geist: ["Geist-Regular"],
        geistBlack: ["Geist-Black"],
        geistBold: ["Geist-Bold"],
        geistLight: ["Geist-Light"],
        geistMedium: ["Geist-Medium"],
        geistSemiBold: ["Geist-SemiBold"],
        geistThin: ["Geist-Thin"],
        geistUltraBlack: ["Geist-UltraBlack"],
        geistUltraLight: ["Geist-UltraLight"],
      },
      colors: {
        border: "hsl(var(--border))",
        input: "hsl(var(--input))",
        ring: "hsl(var(--ring))",
        background: "hsl(var(--background))",
        foreground: "hsl(var(--foreground))",
        primary: {
          DEFAULT: "hsl(var(--primary))",
          foreground: "hsl(var(--primary-foreground))",
        },
        navy: {
          DEFAULT: "hsl(var(--navy))",
          foreground: "hsl(var(--navy-foreground))",
        },
        darkred: {
          DEFAULT: "hsl(var(--darkred))",
          foreground: "hsl(var(--darkred-foreground))",
        },
        secondary: {
          DEFAULT: "hsl(var(--secondary))",
          foreground: "hsl(var(--secondary-foreground))",
        },
        destructive: {
          DEFAULT: "hsl(var(--destructive))",
          foreground: "hsl(var(--destructive-foreground))",
        },
        muted: {
          DEFAULT: "hsl(var(--muted))",
          foreground: "hsl(var(--muted-foreground))",
        },
        accent: {
          DEFAULT: "hsl(var(--accent))",
          foreground: "hsl(var(--accent-foreground))",
        },
        popover: {
          DEFAULT: "hsl(var(--popover))",
          foreground: "hsl(var(--popover-foreground))",
        },
        card: {
          DEFAULT: "hsl(var(--card))",
          foreground: "hsl(var(--card-foreground))",
        },
      },
      borderWidth: {
        hairline: hairlineWidth(),
      },
      keyframes: {
        "accordion-down": {
          from: { height: "0" },
          to: { height: "var(--radix-accordion-content-height)" },
        },
        "accordion-up": {
          from: { height: "var(--radix-accordion-content-height)" },
          to: { height: "0" },
        },
      },
      animation: {
        "accordion-down": "accordion-down 0.2s ease-out",
        "accordion-up": "accordion-up 0.2s ease-out",
      },
    },
  },
  plugins: [require("tailwindcss-animate")],
};
```

---

`~/lib/useColorScheme.tsx`:

```tsx
import { useColorScheme as useNativewindColorScheme } from "nativewind";

export function useColorScheme() {
  const { colorScheme, setColorScheme, toggleColorScheme } =
    useNativewindColorScheme();
  return {
    colorScheme: colorScheme ?? "dark",
    isDarkColorScheme: colorScheme === "dark",
    setColorScheme,
    toggleColorScheme,
  };
}
```

---

`ThemeToggle.tsx`:

```tsx
import { Pressable, View } from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { setAndroidNavigationBar } from "~/lib/android-navigation-bar";
import { MoonStar } from "~/lib/icons/MoonStar";
import { Sun } from "~/lib/icons/Sun";
import { useColorScheme } from "~/lib/useColorScheme";
import { cn } from "~/lib/utils";

export function ThemeToggle() {
  const { isDarkColorScheme, setColorScheme } = useColorScheme();
  return (
    <Pressable
      onPress={() => {
        const newTheme = isDarkColorScheme ? "light" : "dark";
        setColorScheme(newTheme);
        setAndroidNavigationBar(newTheme);
        AsyncStorage.setItem("theme", newTheme);
      }}
      className="web:ring-offset-background web:transition-colors web:focus-visible:outline-none web:focus-visible:ring-2 web:focus-visible:ring-ring web:focus-visible:ring-offset-2"
    >
      {({ pressed }) => (
        <View
          className={cn(
            "flex-1 aspect-square pt-0.5 justify-center items-start web:px-5",
            pressed && "opacity-70"
          )}
        >
          {isDarkColorScheme ? (
            <MoonStar
              className="text-foreground"
              size={23}
              strokeWidth={1.25}
            />
          ) : (
            <Sun className="text-foreground" size={24} strokeWidth={1.25} />
          )}
        </View>
      )}
    </Pressable>
  );
}
```

---

`/app/_layout.tsx:`

```tsx
import "~/global.css";

import { Theme, ThemeProvider } from "@react-navigation/native";
import { Slot, SplashScreen } from "expo-router";
import { StatusBar } from "expo-status-bar";
import { NAV_THEME } from "~/lib/constants";
import { useColorScheme } from "~/lib/useColorScheme";
import { PortalHost } from "@rn-primitives/portal";
import { useEffect, useState } from "react";
import { useFonts } from "expo-font";
import AuthProvider from "~/providers/AuthProvider";
import { GestureHandlerRootView } from "react-native-gesture-handler";
import { setAndroidNavigationBar } from "~/lib/android-navigation-bar";
import AsyncStorage from "@react-native-async-storage/async-storage";

const LIGHT_THEME: Theme = {
  dark: false,
  colors: NAV_THEME.light,
};
const DARK_THEME: Theme = {
  dark: true,
  colors: NAV_THEME.dark,
};

export {
  // Catch any errors thrown by the Layout component.
  ErrorBoundary,
} from "expo-router";

export const unstable_settings = {
  initialRouteName: "/",
};

// Prevent the splash screen from auto-hiding before getting the color scheme.
SplashScreen.preventAutoHideAsync();

export default function RootLayout() {
  const [isColorSchemeLoaded, setIsColorSchemeLoaded] = useState(false);

  const { colorScheme, setColorScheme, isDarkColorScheme } = useColorScheme();

  const [loaded, error] = useFonts({
    "Geist-Black": require("~/assets/fonts/Geist-Black.ttf"),
    "Geist-Bold": require("~/assets/fonts/Geist-Bold.ttf"),
    "Geist-Light": require("~/assets/fonts/Geist-Light.ttf"),
    "Geist-Medium": require("~/assets/fonts/Geist-Medium.ttf"),
    "Geist-Regular": require("~/assets/fonts/Geist-Regular.ttf"),
    "Geist-SemiBold": require("~/assets/fonts/Geist-SemiBold.ttf"),
    "Geist-Thin": require("~/assets/fonts/Geist-Thin.ttf"),
    "Geist-UltraBlack": require("~/assets/fonts/Geist-UltraBlack.ttf"),
    "Geist-UltraLight": require("~/assets/fonts/Geist-UltraLight.ttf"),
  });

  useEffect(() => {
    (async () => {
      const theme = await AsyncStorage.getItem("theme");
      console.log("🚀 ~ INITIAL LOADING THEME:", theme);

      if (!theme) {
        AsyncStorage.setItem("theme", colorScheme);
        setIsColorSchemeLoaded(true);
        return;
      }

      const colorTheme = theme === "dark" ? "dark" : "light";

      if (colorTheme !== colorScheme) {
        setColorScheme(colorTheme);
        setAndroidNavigationBar(colorTheme);
        setIsColorSchemeLoaded(true);
        return;
      }

      setAndroidNavigationBar(colorTheme);
      setIsColorSchemeLoaded(true);
    })().finally(() => {
      SplashScreen.hideAsync();
    });
  }, []);

  if (!isColorSchemeLoaded) {
    return null;
  }

  if (!loaded && !error) {
    return null;
  }

  return (
    <ThemeProvider value={isDarkColorScheme ? DARK_THEME : LIGHT_THEME}>
      <GestureHandlerRootView style={{ flex: 1 }}>
        <AuthProvider>
          <StatusBar style={isDarkColorScheme ? "light" : "dark"} />
          <Slot />
          <PortalHost />
        </AuthProvider>
      </GestureHandlerRootView>
    </ThemeProvider>
  );
}
```

**Comment by periakteon** [comment]
I think it is related to a problem already mentioned in #1003

**Comment by periakteon** [comment]
I fixed the problem by changing

`"nativewind": "^4.0.33",`

to

`"nativewind": "4.0.36",`

**Comment by tyrauber** [comment]
Likewise. Works in `4.0.36`. Fails in `4.1.1`. Issue is specifically `setColorScheme`.

**Comment by danstepanov** [comment]
Please provide a repro in v4.1
https://github.com/nativewind/nativewind/blob/next/contributing.md#opening-an-issue

**Comment by tyrauber** [comment]
Fixed for me in 4.1.4. Thanks.
